src/code.rs

```diff
@@ -422,25 +422,62 @@ impl {struct_name} {{
     ));
 
     buffer.push_str(&format!(r##"
-    /// Paginates through the table where page is a 0-based index (i.e. page 0 is the first page)
-    pub{async_keyword} fn paginate(db: &mut ConnectionType, page: i64, page_size: i64) -> QueryResult<PaginationResult<Self>> {{
+    /// Paginates through the table where page is a 1-based index (i.e. page 1 is the first page)
+    pub{async_keyword} fn paginate(db: &mut ConnectionType, param_page_starting_with_1: i64, param_page_size: i64, filter: {struct_name}Filter) -> QueryResult<PaginationResult<Self>> {{
         use {schema_path}{table_name}::dsl::*;
 
-        let page_size = if page_size < 1 {{ 1 }} else {{ page_size }};
-        let total_items = {table_name}.count().get_result(db){await_keyword}?;
-        let items = {table_name}.limit(page_size).offset(page * page_size).load::<Self>(db){await_keyword}?;
+        let param_page = param_page_starting_with_1.max(0);
```

Comment:
```suggestion
        let param_page = param_page_starting_with_1.max(1);
```

guessing by the previous implementation and the updated doc-comment, shouldnt the lowest be 1 instead of 0?

Reply:
ah, this is actually a bug, it should be:
`let param_page = (page_starting_with_1 - 1).max(0);`

Reply:
ok, changed this: I reverted to a 0-based page API